app/src/main/java/com/group80/uoftinder/UpdateList.java

```diff
@@ -0,0 +1,51 @@
+package com.group80.uoftinder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UpdateList extends User {
+    private User currentUser; // the current user
+    private User displayedUser; // the user that is displayed to current user
+    private boolean liked; // true if currentUser 'likes' displayedUser, false otherwise
+    private List<User> viewedList; // currentUser's viewedList
+    private List<User> likedList; // currentUser's likedList
+
+    /**
+     * Constructor for UpdateList Class.
+     * @param displayedUser is the user that is currently being displayed to currentUser.
+     * @param liked is True when currentUser 'likes' displayedUser, False otherwise.
+     */
+    public UpdateList(User currentUser, User displayedUser, boolean liked) {
+        super();
+        this.currentUser = currentUser;
+        this.displayedUser = displayedUser;
+        this.liked = liked;
+        this.viewedList = currentUser.getViewed();
+        this.likedList = currentUser.getLiked();
+    }
+
+    /**
+     * Adds displayedUser to currentUser's viewed list.
+     * If liked is true, also adds displayedUser to currentUser's liked list.
+     * @param displayedUser is the user that is currently being displayed to currentUser.
+     * @param liked is True when currentUser 'likes' displayedUser, False otherwise.
+     */
+    public void addToList(User displayedUser, boolean liked) {
```

Comment:
From what it seems here, a new instance of UpdateList needs to be created per displayedUser. I suggest making this method a static method so it's easier to call each time there is a new displayedUser.